docs/spec_draft.md

```diff
@@ -927,6 +928,47 @@ spaces of `result` and `operand`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.reverse
+
+### Semantics
+
+Reverses the order of elements in the `operand` along the specified `dimensions`, generating an output array of the same shape. Each element of the operand array at a multidimensional index is stored into the output array at a transformed index. The multidimensional index is transformed by reversing the index in each dimension to be reversed (i.e., if a dimension of size N is one of the reversing dimensions, its index i is transformed into N - 1 - i).
+
+### Operands
+
+| Name         | Type                               |
+|--------------|------------------------------------|
+| `operand`    | tensor of any supported types      |
+| `dimensions` | 1-dimensional array of type `si64` |
+
+### Results
+
+| Name     | Type                          |
+|----------|-------------------------------|
+| `result` | tensor of any supported types |
+
+### Constraints
+
+  * (C1) `operand` and `result` have the same element type.
+  * (C2) `operand` and `result` have the same shape.
+
+### Examples
+
+```mlir
+// %operand = [[1, 2], [3, 4], [5, 6]]
+%result = "stablehlo.reverse"(%operand) {dimensions = dense<[0]> : tensor<1xi64>} : (tensor<3x2xi32>) -> tensor<3x2xi32>
```

Comment:
Can be simplified to `dense<0>`. Same below.

Reply:
On a related note, can be simplified to `tensor<i64>` for scalars.

Nit: add space between the attribute and the curly braces?
```
%result = "stablehlo.reverse"(%operand) { dimensions = dense<0> : tensor<i64> } : (tensor<3x2xi32>) -> tensor<3x2xi32>
```

Reply:
Good idea about `tensor<i64>`! Didn't know that would work too.

Re: spacing, that would work too, but that's not the canonical form of the MLIR generic syntax (the canonical form doesn't have spacing).

Reply:
Re: spacing, does the canonical form allow multi-line? This would be hard to follow on examples with bunch of attributes like `gather`.

Reply:
Oof you're right I didn't consider that.

I think that the canonical form (i.e. what's printed by -opt tools) just prints everything on a single line. I agree that this is hard to follow. 

This is why personally I've been using [a notation like this](https://github.com/tensorflow/mlir-hlo/blob/34028950c5fac34a074bb9da63659acea210bbcb/tests/Dialect/mhlo/hlo-legalize-to-stablehlo.mlir#L902-L911), but that's not standardized. What do you think about that format?

Reply:
No objections for the format, I was implicitly following that format as well for ops that are too long to write in one line.

I'll also create a PR fixing existing ops with attributes to follow this format.